test/test_torch.py

```diff
@@ -1242,6 +1242,45 @@ def backward_func(slf, device):
 
         backward_func(self, device)
 
+    @dtypes(*floating_types())
+    @onlyNativeDeviceTypes
+    def test_nondeterministic_alert_MaxUnpool1d(self, device, dtype):
+        module = torch.nn.MaxUnpool1d(3, 1)
+        input = torch.randn(1, 1, 7, dtype=dtype, device=device)
+        indices = torch.zeros_like(input, dtype=torch.long, device=device)
+
+        @expectedAlertNondeterministic('max_unpooling2d_forward_out')
+        def forward_func(slf, device):
+            module(input, indices)
+
+        forward_func(self, device)
+
+    @dtypes(*floating_types())
+    @onlyNativeDeviceTypes
+    def test_nondeterministic_alert_MaxUnpool2d(self, device, dtype):
+        module = torch.nn.MaxUnpool2d(3, 1)
+        input = torch.randn(1, 1, 7, 7, dtype=dtype, device=device)
+        indices = torch.zeros_like(input, dtype=torch.long, device=device)
+
+        @expectedAlertNondeterministic('max_unpooling2d_forward_out')
+        def forward_func(slf, device):
+            module(input, indices)
+
+        forward_func(self, device)
+
+    @dtypes(*floating_types())
+    @onlyNativeDeviceTypes
+    def test_nondeterministic_alert_MaxUnpool3d(self, device, dtype):
+        module = torch.nn.MaxUnpool3d(3, 1)
+        input = torch.randn(1, 1, 7, 7, 7, dtype=dtype, device=device)
+        indices = torch.zeros_like(input, dtype=torch.long, device=device)
+
+        @expectedAlertNondeterministic('max_unpooling3d_forward_out')
+        def forward_func(slf, device):
+            module(input, indices)
+
+        forward_func(self, device)
```

Comment:
Just curious. Does `forward_func` use `self/device` at all?

Reply:
No, but they're required for the decorator. #84808 will make these tests easier to read